### DISCUSSION TITLE
Simple VOL_UP/VOL_DN combo not working

### DISCUSSION THREAD

**Discussion by BarkMulcher** [nickcoutsos/keymap-editor#275] [Q&A] [answered]
Hello all. Thank you to the dev for making this GUI. Hella sick. I'm brand new to ZMK and keyboard customization in general. I have a Kinesis Advantage360. For my first combo, I'm trying to add a shortcut for volume up/volume down on the Win+PgUp/PgDn keys. I followed the instructions in the manual to a 'T', flashed the firmware to the separated boards' drives, but neither combo is working. Since it's such a (seemingly) simple combo, I'm not sure where to start troubleshooting. This isn't mentioned in the manual, but it occurred to me to ask whether I should have deleted the existing .uf2 files in the drives before copying the updated firmware files. I've attached a screenshot of the keys selected in the GUI. Any advice or direction would be massively appreciated! 

edit: I have restarted and completed the process two more times to verify that I'm doing it per the instructions. I notice that when I copy over the new .uf2 file into the respective drive, the keyboard flashes lights and auto-disconnects as described in the manual, but when I re-open the drive, the old firmware named `CURRENT.uf2` is still present and the only .uf2 file in the drive, with a modified date of 06/30/2024 (not the current date of modification). This makes me wonder if the flashing process is even happening. I wonder if someone could verify that this is the expected behavior. If it is, then maybe this key combo is incompatible? I will try a different key combo without the Win key and report back.

edit2: re-tried the process with the combo `fn+PgUp` and `fn+PgDn`. Still nothing. 

![Screenshot 2024-12-23 140057](https://github.com/user-attachments/assets/c4c353e1-1e79-4a41-8ce4-df94b706aeec)

**Comment by nickcoutsos** [comment, marked as answer]
Looks ok to me... I'd want to rule a couple of things out first:

1. I notice the combo keys are both on the right half of the keyboard; any chance you're only flashing the firmware to the right half? All inputs are going to be processed on the left so make sure you're flashing firmware changes there
2. Some people get the wrong idea about combos and you mention one of the keys as <kbd>fn</kbd>... all keys in the combo need to be pressed at approximately the same time (it's configurable, but 50ms by default) in contrast to something traditional combos like <kbd>Ctrl</kbd>+<kbd>C</kbd> where one key is held while the other is pressed. If you _do_ want the latter behavior you might actually prefer moving the binding to a layer.

**Reply by BarkMulcher** [reply]
Yooo thank you so much, Nick! For both the prompt reply and a resolution! # 2 was the answer, and makes a lot of sense. Pressing fn+pgup/dn at the same time causes the intended behavior. I will look into moving the binding to a layer. Where would I read up on doing that correctly? But this will work for now. Thank you again.

**Reply by nickcoutsos** [reply]
Hey, nice!

If you want to read about layers, here: https://zmk.dev/docs/keymaps/behaviors/layers

The short version is having a key bound to `&mo` (momentarily activate a layer) -- I think `fn` is already doing this -- and then on the target layer having the desired actions bound to the keys you wanted to tap.
